addons/web/static/src/core/dialog/dialog.js

```diff
@@ -74,6 +72,10 @@ export class Dialog extends Component {
         this.position.left = 0;
         this.position.top = 0;
     }
+
+    close() {
+        this.data.close();
+    }
```

Comment:
I prefer naming it `onEscape`, just so we know in the override that it customises the behaviour of pressing `Esc` on keyboard.

Override on `close()` gives the wrong impression that we cannot close the dialog of `mail.compose.message` at all, including with buttons.